src/metabase/models/revision.clj

```diff
@@ -134,19 +135,21 @@
          (integer? id)
          (t2/exists? entity :id id)
          (map? object)]}
-  (let [object (serialize-instance entity id (dissoc object :message))]
+  (let [serialized-object (serialize-instance entity id (dissoc object :message))
+        last-object       (t2/select-one-fn :object Revision :model (name entity) :model_id id {:order-by [[:id :desc]]})]
```

Comment:
This works, but it's a waste of a DB call if we already know that the object has changed or not further up the call tree. For example we should be able to skip this check if `is-creation?` is true, or if it's a delete event. I also checked the call sites for `:dashboard-update`, `:segment-update`, and `:metric-update`, and it seems like we should be able to calculate whether or not anything has actually changed there too.